File: LeetCode/cpp/1700_Number_of_Students_Unable_to_Eat_Lunch.cpp
class Solution {
public:
	int countStudents(vector<int>& students, vector<int>& sandwiches) {
		int answer = 0;
		queue<int> studentsQ, sandwichesQ;

		for (int student : students) {
			studentsQ.push(student);
		}

		for (int sandwich : sandwiches) {
			sandwichesQ.push(sandwich);
		}

		int cnt = 0;
		while (!studentsQ.empty() && !sandwichesQ.empty()) {
			if (studentsQ.front() == sandwichesQ.front()) {
				studentsQ.pop();
				sandwichesQ.pop();
				cnt = 0;
			}
			else {
				int n = studentsQ.front();
				studentsQ.pop();
				studentsQ.push(n);
				cnt++;

				if (cnt > students.size()) {
					break;
				}
			}
		}

		if (!studentsQ.empty()) {
			answer = studentsQ.size();
		}

		return answer;
	}
};

File: LeetCode/java/1700_Number_of_Students_Unable_to_Eat_Lunch.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int answer = 0;
        Queue<Integer> studentsQ = new LinkedList<>();
        Queue<Integer> sandwichesQ = new LinkedList<>();

        for (int student : students) {
            studentsQ.add(student);
        }

        for (int sandwich : sandwiches) {
            sandwichesQ.add(sandwich);
        }

        int cnt = 0;
        while (!studentsQ.isEmpty() && !sandwichesQ.isEmpty()) {
            if (studentsQ.peek() == sandwichesQ.peek()) {
                studentsQ.poll();
                sandwichesQ.poll();
                cnt = 0;
            } else {
                int n = studentsQ.poll();
                studentsQ.add(n);
                cnt++;

                if (cnt > students.length) {
                    break;
                }
            }
        }

        if (!studentsQ.isEmpty()) {
            answer = studentsQ.size();
        }

        return answer;
    }
}